includes/islandora_batch_object_base.inc

```diff
@@ -71,6 +72,34 @@ abstract class IslandoraBatchObject extends IslandoraNewFedoraObject {
    */
   abstract public function addRelationships();
 
+  /**
+   * The actual back-end action that is run after the batchProcess.
+   *
+   * Classes not overriding this will get objects pushed to the back-end.
+   * (ingested)
+   *
+   * @return AbstractObject|false
+   *   Either a newly ingested Abstract Object or false if failed.
+   */
+  public function batchRepositoryAction() {
+    if (!islandora_object_load($this->id)) {
+      return islandora_add_object($this);
+    }
+    else {
+      return FALSE;
+    }
+  }
+
+  /**
+   * A human readable short description of the repository action.
+   *
+   * @return string
+   *   Recomended value is a verb in past tense.
+   */
+  public function getRepositoryActionDescription() {
+    return "ingested";
```

Comment:
Would that mean generating full messages for success, exceptions and errors? Issue i see with that is that adding then afterwards arguments coming from the actual batch would make things quite complicated.  What do you propose

Reply:
Ok, will work on the template strings idea, probably 3 ones would suffice. Wonder in that case if the full generation t()  et all // - hate having t()in a class...-  + argument expansion/replacement should happen an implementing class or should stay on batch itself as it is today? ideas @d-r-p @adam-vessey @bryjbrown ? The simpler the better i would say because the more i add the more i will be open other wormholes .. the implementation would be aware that batch itself only provides the %pid really...and some Drupal exception message...